crates/cubecl-wgpu/src/backend/vulkan.rs

```diff
@@ -0,0 +1,275 @@
+    }
+
+    if let Some(float_controls2) = &extended_feat.float_controls2 {
+        if float_controls2.shader_float_controls2 == TRUE {
+            comp_options.supports_fp_fast_math = true;
+        }
+    }
+
+    if extended_feat.cmma.is_some() {
+        register_cmma(ash, adapter, props);
+    }
+}
+
+fn register_types(props: &mut DeviceProperties<Feature>, ext_feat: &ExtendedFeatures<'_>) {
+    use cubecl_core::ir::{Elem, FloatKind, IntKind};
+
+    let mut register = |elem| {
+        props.register_feature(Feature::Type(elem));
+    };
+
+    let supported_types = [
+        Elem::UInt(UIntKind::U16),
+        Elem::UInt(UIntKind::U32),
+        Elem::UInt(UIntKind::U64),
+        Elem::Int(IntKind::I16),
+        Elem::Int(IntKind::I32),
+        Elem::Int(IntKind::I64),
+        Elem::AtomicInt(IntKind::I32),
+        Elem::AtomicInt(IntKind::I64),
+        Elem::AtomicUInt(UIntKind::U32),
+        Elem::AtomicUInt(UIntKind::U64),
+        Elem::Float(FloatKind::F32),
+        //Elem::Float(FloatKind::F64),
+        Elem::Bool,
+    ];
+
+    for ty in supported_types {
+        register(ty);
+    }
+
+    if ext_feat.float16_int8.shader_float16 == TRUE {
+        register(Elem::Float(FloatKind::F16));
+    }
+    if ext_feat.float16_int8.shader_int8 == TRUE {
+        register(Elem::Int(IntKind::I8));
+        register(Elem::UInt(UIntKind::U8));
+    }
+
+    if let Some(atomic_float) = ext_feat.atomic_float {
+        if atomic_float.shader_buffer_float32_atomics == TRUE {
+            register(Elem::AtomicFloat(FloatKind::F32));
+            register(Elem::AtomicFloat(FloatKind::F64));
+        }
+    }
+    if let Some(atomic_float) = ext_feat.atomic_float2 {
+        if atomic_float.shader_buffer_float16_atomics == TRUE {
+            register(Elem::AtomicFloat(FloatKind::F16));
+        }
+    }
+}
+
+fn register_cmma(
+    ash: &InstanceShared,
+    adapter: &vulkan::Adapter,
+    props: &mut DeviceProperties<Feature>,
+) {
+    let cmma = cooperative_matrix::Instance::new(ash.entry(), ash.raw_instance());
+    let properties = unsafe {
+        cmma.get_physical_device_cooperative_matrix_properties(adapter.raw_physical_device())
+            .unwrap()
+    };
+    let sizes = properties
+        .into_iter()
+        .filter(|it| {
+            it.saturating_accumulation == 0
+                && it.result_type == it.c_type
+                && it.scope == ScopeKHR::SUBGROUP
+        })
+        .filter_map(|it| {
+            Some(Feature::Cmma {
+                a: conv_type(it.a_type)?,
+                b: conv_type(it.b_type)?,
+                c: conv_type(it.c_type)?,
+                m: it.m_size as u8,
+                k: it.k_size as u8,
+                n: it.n_size as u8,
+            })
+        })
+        .collect::<Vec<_>>();
+    log::debug!("Supported CMMA sizes: {sizes:#?}");
+    for size in sizes {
+        props.register_feature(size);
+    }
+}
+
+fn conv_type(vk_ty: ComponentTypeKHR) -> Option<Elem> {
```

Comment:
conv => comp ?

Reply:
conv as in convert, I should spell it out